src/allotropy/parsers/revvity_kaleido/kaleido_structure.py

```diff
@@ -0,0 +1,369 @@
+    "S": SampleRoleType.standard_sample_role.value,
+    "U": SampleRoleType.unknown_sample_role.value,
+    "E": SampleRoleType.control_sample_role.value,
+    "ZL": SampleRoleType.control_sample_role.value,
+    "ZH": SampleRoleType.control_sample_role.value,
+    "LB": SampleRoleType.control_sample_role.value,
+    "LC": SampleRoleType.control_sample_role.value,
+    "LH": SampleRoleType.control_sample_role.value,
+}
+
+
+SCAN_POSITION_CONVERSION = {
+    "TOP": ScanPositionSettingPlateReader.top_scan_position__plate_reader_,
+    "BOTTOM": ScanPositionSettingPlateReader.bottom_scan_position__plate_reader_,
+}
+
+
+TRANSMITTED_LIGHT_CONVERSION = {
+    "BRIGHTFIELD": TransmittedLightSetting.brightfield,
+    "DARKFIELD": TransmittedLightSetting.darkfield,
+    "PHASE CONTRAST": TransmittedLightSetting.phase_contrast,
+}
+
+
+@dataclass
+class BackgroundInfo:
+    experiment_type: str
+
+
+@dataclass(frozen=True)
+class Results:
+    barcode: str
+    data: dict[str, str]
+
+    def get_plate_well_count(self) -> int:
+        return len(self.data)
+
+    def get_well_float_value(self, well_position: str) -> float:
+        return try_float(
+            self.get_well_str_value(well_position),
+            f"result well at '{well_position}'",
+        )
+
+    def get_well_str_value(self, well_position: str) -> str:
+        return str(
+            assert_not_none(
+                self.data.get(well_position),
+                msg=f"Unable to get well at position '{well_position}' from results section.",
+            )
+        )
+
+
+@dataclass(frozen=True)
+class AnalysisResult:
+    analysis_parameter: str
+    results: dict[str, str]
+
+    @staticmethod
+    def create(reader: CsvReader) -> Optional[AnalysisResult]:
+        barcode_line = assert_not_none(
+            reader.drop_until_inclusive("^Barcode:(.+),"),
+            msg="Unable to find background information.",
+        )
+
+        analysis_parameter = None
+        for element in barcode_line.split(","):
+            if re.search("^.+:.+$", element):
+                key, value = element.split(":", maxsplit=1)
+                if "AnalysisParameter" in key:
+                    analysis_parameter = value
+                    break
+
+        analysis_parameter = assert_not_none(
+            analysis_parameter,
+            msg="Unable to find analysis parameter in Analysis Results section.",
+        )
+
+        try:
+            results = assert_not_none(
+                reader.pop_csv_block_as_df(header=0, index_col=0),
+                msg="Unable to find results table.",
+            )
+        except AllotropeConversionError:
+            logging.warning(
+                f"Unable to read analysis result '{analysis_parameter}'. Ignoring"
+            )
+            return None
+
+        for column in results:
+            if str(column).startswith("Unnamed"):
+                results = results.drop(columns=column)
+
+        try:
```

Comment:
Can you add a comment explaining what this is going, specifically, the `try_float_or_none` check.